programs/lz4io.c

```diff
@@ -472,7 +531,12 @@ static int LZ4IO_compressFilename_extRess(cRess_t ress, const char* srcFileName,
     /* single-block file */
     if (readSize < blockSize) {
         /* Compress in single pass */
-        size_t const cSize = LZ4F_compressFrame(dstBuffer, dstBufferSize, srcBuffer, readSize, &prefs);
+        size_t cSize;
+        if (ress.cdict) {
```

Comment:
minor : `LZ4F_compressFrame_usingCDict()` with a `NULL` cdict as argument is exactly the same as `LZ4F_compressFrame()` : 
https://github.com/lz4/lz4/blob/dev/lib/lz4frame_static.h#L102
It's implemented this way by the way : 
https://github.com/lz4/lz4/blob/dev/lib/lz4frame.c#L385

So it allows you to simplify this branch.